eclair-core/src/main/scala/fr/acinq/eclair/wire/LightningMessageCodecs.scala

```diff
@@ -286,4 +289,20 @@ object LightningMessageCodecs {
     .typecase(264, replyChannelRangeCodec)
     .typecase(265, gossipTimestampFilterCodec)
 
+  val meteredLightningMessageCodec = Codec[LightningMessage](
+    (msg: LightningMessage) => KamonExt.time("scodec.encode.time", tags = TagSet.of("type", msg.getClass.getSimpleName))(lightningMessageCodec.encode(msg)),
+    (bits: BitVector) => {
+      // this is a bit more involved, because we don't know beforehand what the type of the message will be
+      val timer = Kamon.timer("scodec.decode.time")
+      val begin = System.nanoTime()
+      val res = lightningMessageCodec.decode(bits)
+      val end = System.nanoTime()
+      res match {
+        case Attempt.Successful(decoded) => timer.withTag("type", decoded.value.getClass.getSimpleName).record(end - begin)
+        case Attempt.Failure(_) => timer.withTag("type", "unknown").record(end - begin)
+      }
+      res
+    }
+  )
```

Comment:
LGTM but what is the performance cost of adding timers here ?

Reply:
I don't know, the assumption is that the overhead of metrics is not significant compared to the code being measured.

According to the [doc](https://kamon.io/docs/latest/core/metrics/) I should probably statically define the metrics:

> When possible, define and refine your metrics up front, as a static member or instance members on the instrumented components of your services. This saves a bit of time in looking up instruments. If you can’t do so, rest assured that Kamon will always return the same instrument instance if requested with the same arguments (metric name and tags) from anywhere in your code.